File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.go
func sortVowels(s string) string {
    vowels := "aeiouAEIOU"
	vowelIndices := []int{}
	vowelChars := []byte{}

	// Identify vowel indices and collect vowel characters
	for i, char := range s {
		if strings.ContainsRune(vowels, char) {
			vowelIndices = append(vowelIndices, i)
			vowelChars = append(vowelChars, byte(char))
		}
	}

	// Sort the vowel characters in non-decreasing order
	sort.Slice(vowelChars, func(i, j int) bool {
		return vowelChars[i] < vowelChars[j]
	})

	// Create the resulting string t
	t := []byte(s)
	for i, index := range vowelIndices {
		t[index] = vowelChars[i]
    }

	return string(t)
}

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.cs
public class Solution {
    public string SortVowels(string s) {
        string vowels = "aeiouAEIOU";
        int[] vowelIndices = new int[s.Length];
        char[] vowelChars = new char[s.Length];
        int vowelCount = 0;

        // Identify vowel indices and collect vowel characters
        for (int i = 0; i < s.Length; i++)
        {
            if (vowels.Contains(s[i]))
            {
                vowelIndices[vowelCount] = i;
                vowelChars[vowelCount] = s[i];
                vowelCount++;
            }
        }

        // Sort the vowel characters in non-decreasing order
        Array.Sort(vowelChars, 0, vowelCount);
        
        // Create the resulting string t
        char[] t = s.ToCharArray();
        for (int i = 0; i < vowelCount; i++)
        {
            t[vowelIndices[i]] = vowelChars[i];
        }

        return new string(t);
    }
}